plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,321 @@
+                // By default, the clickhouse column is not allowed to be null
+                clickhouseColumns.add(column.replace(" NOT NULL", ""));
+            }
+            else {
+                String name = column.split(" ")[0];
+                String type = column.substring(name.length() + 1);
+                clickhouseColumns.add(String.format("%s Nullable(%s)", name, type));
+            }
+        }
+        ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+        ClickHouseTableProperties.getEngine(tableMetadata.getProperties())
+                .ifPresent(value -> tableOptions.add("ENGINE =" + value));
+        return format("CREATE TABLE %s (%s) %s", quoted(remoteTableName), join(", ",
+                clickhouseColumns), join(", ", tableOptions.build()));
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "CREATE DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "DROP DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    protected Optional<BiFunction<String, Long, String>> limitFunction()
+    {
+        return Optional.of((sql, limit) -> sql + " LIMIT " + limit);
+    }
+
+    @Override
+    protected ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        // ClickHouse does not have schemas
+        DatabaseMetaData metadata = connection.getMetaData();
+        return metadata.getTables(
+                null,
+                schemaName.orElse(null),
+                escapeNamePattern(tableName, metadata.getSearchStringEscape()).orElse(null),
+                new String[] {"TABLE", "VIEW"});
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn,
+            String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format("ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getRemoteTableName()),
+                    quoted(jdbcColumn.getColumnName()), quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName,
+            SchemaTableName newTable)
+    {
+        String sql = format("RENAME TABLE %s.%s TO %s.%s", schemaName, tableName, newTable.getSchemaName(),
```

Comment:
Are the table and database names not quoted here intentionally? Other methods are quoting the table name.